src/umock_lock_factory_pthread.c

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <pthread.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "umock_c/umockalloc.h"
+#include "umock_c/umock_log.h"
+#include "umock_c/umock_lock_factory.h"
+#include "umock_c/umock_lock_factory_default.h"
+
+typedef struct UMOCK_C_LOCK_PTHREADS_TAG
+{
+    UMOCK_C_LOCK_IF lock_if;
+    pthread_rwlock_t rw_lock;
+} UMOCK_C_LOCK_PTHREADS;
+
+static void umock_lock_pthreads_acquire_shared(UMOCK_C_LOCK_HANDLE lock)
+{
+    if (lock == NULL)
+    {
+        /* Codes_SRS_UMOCK_LOCK_FACTORY_PTHREAD_01_010: [ If lock is NULL, umock_lock_pthread_acquire_shared shall return. ]*/
+        UMOCK_LOG("Invalid arguments: UMOCK_C_LOCK_HANDLE lock=%p", lock);
+    }
+    else
+    {
+        UMOCK_C_LOCK_PTHREADS* umock_c_lock_windows = (UMOCK_C_LOCK_PTHREADS*)lock;
```

Comment:
>windows [](start = 44, length = 7)

not windows here

Reply:
Fixed.

---
In reply to: [530652478](https://github.com/Azure/umock-c/pull/177#discussion_r530652478) [](ancestors = 530652478)